### PR TITLE
Added HttpModule the option to set default headers, for all outgoing requests.

### PR DESCRIPTION
**Pull request by MatanKoby**

### CODE REVIEW

File: src/main/java/com/nu/art/http/HttpModule.java [hunk 1]
```diff
@@ -187,6 +188,18 @@ public BaseTransaction() {
 		}
 	}
 
+	private HashMap<String, Getter<String>> defaultHeaders;
+
+	public void addDefaultHeader(String key, Getter<String> value) {
+		if (defaultHeaders == null)
```
Comment (TacB0sS) [inline src/main/java/com/nu/art/http/HttpModule.java]: No need for this condition.. create the instance at the declaration

**Review by TacB0sS — CHANGES_REQUESTED** [review]

File: src/main/java/com/nu/art/http/HttpModule.java [hunk 2]
```diff
@@ -217,7 +230,12 @@ long getTotalHoopTime() {
 		private HoopTiming hoop = new HoopTiming();
 
 		protected IHttpRequest createRequest() {
-			return new HttpRequestIn();
+			HttpRequestIn httpRequest = new HttpRequestIn();
+			if (defaultHeaders != null & defaultHeaders.size() > 0)
```
Comment (TacB0sS) [inline src/main/java/com/nu/art/http/HttpModule.java]: This condition is also gone once created the instance at the declaration

Comment (MatanKoby) [reply src/main/java/com/nu/art/http/HttpModule.java]: Removed the null check.

File: src/main/java/com/nu/art/http/HttpModule.java (hunk 1 above)
Comment (MatanKoby) [reply src/main/java/com/nu/art/http/HttpModule.java]: Added map creation at declaration, and removed the null check.

File: src/main/java/com/nu/art/http/HttpModule.java [hunk 3]
```diff
@@ -217,7 +228,13 @@ long getTotalHoopTime() {
 		private HoopTiming hoop = new HoopTiming();
 
 		protected IHttpRequest createRequest() {
-			return new HttpRequestIn();
+			HttpRequestIn httpRequest = new HttpRequestIn();
+
+			if (defaultHeaders.size() > 0) // If exist, add default headers to all requests.
```
Comment (TacB0sS) [inline src/main/java/com/nu/art/http/HttpModule.java]: Unnecessary check

**Review by TacB0sS — APPROVED** [review]

File: src/main/java/com/nu/art/http/HttpModule.java (hunk 3 above)
Comment (MatanKoby) [reply src/main/java/com/nu/art/http/HttpModule.java]: Removed the size check.

**Pull request merged**